openapi_configuration.yml

```diff
@@ -540,41 +540,199 @@ paths:
+        Return the best-scoring extraction from the configurations in the document type, or from the
+        specified configuration. # question: 'specified config' true? I was having trouble with the stringified json so didn't test successfully
+      tags:
+      - Reference document exraction
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/PostGoldenExtraction'
+      responses:
+          '200':
+            content:
+              application/json:
+                schema:
+                  $ref: '#/components/schemas/ResponseGoldenExtraction'
+            description: The structured data extracted from the reference document 
+
+          400: # todo I didn't verify 40x responses for any extract_from_goldens, just assumed
+            $ref: '#/components/responses/400'
+          401:
+            $ref: '#/components/responses/401'
+          404:
+            $ref: '#/components/responses/404'
+          415:
+            $ref: '#/components/responses/415'
+          500:
+            $ref: '#/components/responses/500'
+
+  /extract_from_golden/{type}/{name}: # question we call it {name} in other endpoints and {configuration} in other endpoints in serverless.yml. I went with {name} here but actually would prefer to rename as {configuration-name}
+    parameters:
+      - $ref: '#/components/parameters/documentTypeName'
+      - $ref: '#/components/parameters/configurationName'
+      - $ref: '#/components/parameters/environment' 
+    post: 
+      operationId: extract-from-reference-document-and-config
+      summary: Extract from a reference document using the specified published configuration. 
+      description: Extract from a reference document using the specified published configuration
+      tags:
+      - Reference document exraction
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/PostGoldenExtraction'
+      responses:
+          '200':
+            content:
+              application/json:
+                schema:
+                  $ref: '#/components/schemas/ResponseGoldenExtraction'
+            description: The structured data extracted from the reference document
+
+          400: # todo I didn't verify 40x responses for any extract_from_goldens, just assumed
+            $ref: '#/components/responses/400'
+          401:
+            $ref: '#/components/responses/401'
+          404:
+            $ref: '#/components/responses/404'
+          415:
+            $ref: '#/components/responses/415'
+          500:
+            $ref: '#/components/responses/500' 
+
+  /extract_from_golden/{type}/{name}/{version}/{document-name}: # question this one is pretty confusing to me. why is the golden in the path instead of body like the others? why is it a get not post? I didn't dig deep
```

Comment:
sorry that it's a bit inconsistent. we needed a URL that disambiguates from other URLs, and due to that need we have all parameters in the URL, with no request body required (or indeed allowed)

just to keep it clear
"name" refers to the configuration name here
"version" is the version ID of the configuration
"document-name" refers to the golden

and you are free to call these whatever you'd like. they have to be consistent in our code/configuration but that's purely internal